src/translate.rs

```diff
@@ -1162,38 +1180,77 @@ impl FromGlib<i32> for Option<u64> {
     }
 }
 
-/// Translate from a pointer type without taking ownership, transfer: none.
+/// Translate from a pointer type which is annotated with `transfer none`.
+/// The resulting value is referenced at least once, by the bindings.
+///
+/// This is suitable for floating references, which become strong references.
+/// It is also suitable for acquiring non-gobject values, like `gchar*`.
+///
+/// The implementation of this trait should acquire a reference to the value
+/// in a way appropriate to the type, e.g. by referencing or copying.
+/// Values obtained using this trait must be properly released on `drop()`
+/// by the implementing type.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrNone<P: Ptr>: Sized {
     unsafe fn from_glib_none(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type taking ownership, transfer: full.
+/// Translate from a pointer type which is annotated with `transfer full`.
+/// This transfers the ownership of the value to the Rust side.
+///
+/// Because ownership can only be transferred if something is already referenced,
+/// this is unsuitable for floating references.
+///
+/// The implementation of this trait should not acquire any references to the value.
```

Comment:
`acquire references` -> `increase reference count`, and maybe also mention that it should not copy

Reply:
What is meant by "copy" here? Is it about the `Copy` trait or something else?

Reply:
Copy as in creating a new version of the value. Allocating new memory and creating the same thing again. E.g. what the `Clone` trait does on `String`, but not what it does on `Arc`

Reply:
I used "acquire reference" throughout the text as a compromise between "increase reference count" and "copy", because the same trait applies to either kind of data if I understood this right.

Reply:
Yes that's correct. `acquire reference` is not very descriptive though, I would simply mention both possibilities there.

Reply:
Addressed by spelling out behaviour for references and copies.